controllers/storagecluster/storageclient.go

```diff
@@ -47,7 +48,13 @@ func (s *storageClient) ensureCreated(r *StorageClusterReconciler, storagecluste
 	storageClient.Name = storagecluster.Name
 	_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageClient, func() error {
 		if storageClient.Status.ConsumerID == "" {
-			token, err := util.GenerateClientOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath, nil, storagecluster.UID)
+			token, err := util.GenerateClientOnboardingToken(
+				tokenLifetimeInHours,
+				onboardingPrivateKeyFilePath,
+				nil,
+				storagecluster.UID,
+				uuid.New().String(),
+			)
```

Comment:
Reading the secret of the local Consumer for the token instead of generating it here will come in an upcoming PR?

Reply:
yes.